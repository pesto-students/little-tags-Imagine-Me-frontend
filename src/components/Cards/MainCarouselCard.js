

export default function MainCarouselCard({ title, description, image }) {
    console.log(image);
    return <div className="flex h-full relative">
        <div className="absolute inset-0 p-3 flex items-center lg:justify-center lg:relative lg:w-2/4 lg:p-0">
            <div className="w-9/12 md:w-2/4">
                <div className="text-gray-500 text-xl">
                    {title}
                </div>  
                <div className="text-5xl lg:text-6xl">
                    {description}
                </div>
            </div>
        </div>
        <div className="w-full bg-top bg-cover bg-no-repeat lg:w-2/4" style={{backgroundImage: `url(${process.env.REACT_APP_BASE_URL}/${image})`}}>

        </div>
    </div>
}