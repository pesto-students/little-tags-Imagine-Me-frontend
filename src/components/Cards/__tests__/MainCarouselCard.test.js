import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { IntlProvider } from 'react-intl';

import MainCarouselCard from '../MainCarouselCard';

import translation_en from '../../../translations/en.json';

Enzyme.configure({ adapter: new Adapter() });

const { mount } = Enzyme;

describe('<CategoryCarouselCard />', () => {
  it('should renders correctly (snapshot)', () => {
    const tree = mount(
      <IntlProvider locale="en" messages={translation_en}>
        <MainCarouselCard
          title="Trending"
          description="Wear Trending Collections"
          image="image"
        />
      </IntlProvider>,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
