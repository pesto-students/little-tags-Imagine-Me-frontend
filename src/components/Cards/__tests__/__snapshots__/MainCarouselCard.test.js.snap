// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCarouselCard /> should renders correctly (snapshot) 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={
    Object {
      "Bags": "Bags",
      "Hats": "Hats",
      "Jeans": "Jeans",
      "Laptop": "Laptop",
      "T-Shirt": "T-Shirt",
      "Trending": "Trending",
      "Wear Trending Collections": "Wear Trending Collections",
      "cart": "cart",
      "explore": "explore",
      "login": "login",
      "loginNow": "Login Now",
      "loginToStart": "Get started by logging in.",
      "register": "Register",
      "registerNow": "Register Now",
      "registerToStart": "Get started by Registering yourself.",
      "signInInstead": "Sign In Instead",
      "signUpInstead": "Sign Up Instead",
      "trending": "trending",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <MainCarouselCard
    description="Wear Trending Collections"
    image="image"
    title="Trending"
  >
    <div
      className="flex flex-row-reverse h-full relative"
    >
      <div
        className="w-full relative bg-top bg-cover bg-no-repeat lg:w-2/4"
        style={
          Object {
            "backgroundImage": "url(http://127.0.0.1:3001/image)",
          }
        }
      >
        <div
          className="absolute inset-0 bg-gradient-to-r from-gray-300 via-gray-200 to-gray-100 opacity-50 lg:hidden"
        />
      </div>
      <div
        className="absolute inset-0 p-3 flex items-center lg:justify-center lg:relative lg:w-2/4 lg:p-0"
      >
        <div
          className="w-9/12 md:w-2/4"
        >
          <div
            className="text-gray-500 text-xl"
          >
            <FormattedMessage
              id="Trending"
            >
              Trending
            </FormattedMessage>
          </div>
          <div
            className="text-5xl mt-5 lg:text-6xl"
          >
            <FormattedMessage
              id="Wear Trending Collections"
            >
              Wear Trending Collections
            </FormattedMessage>
          </div>
        </div>
      </div>
    </div>
  </MainCarouselCard>
</IntlProvider>
`;
